pixy/consumer.go

```diff
@@ -545,13 +547,32 @@ func (ec *exclusiveConsumer) run() {
 			}
 			select {
 			case ec.tc.messages() <- consumerMessage:
-				pom.CommitOffset(consumerMessage.Offset+1, "")
+				lastSubmittedOffset = consumerMessage.Offset + 1
+				pom.SubmitOffset(lastSubmittedOffset, "")
 			case <-ec.stoppingCh:
-				return
+				goto done
```

Comment:
No, `break` won't work, because it would break out of `select` rather than `for` ([reference](https://golang.org/ref/spec#Break_statements)).